Jenkinsfile

```diff
@@ -426,30 +440,30 @@ def doCheckValgrind() {
     return {
         rlmNode('docker') {
             getArchive()
-            def buildEnv = docker.build 'realm-core-linux:21.04'
+
             def environment = environment()
             environment << 'UNITTEST_NO_ERROR_EXITCODE=1'
-            withEnv(environment) {
-                buildEnv.inside {
-                    def workspace = pwd()
+
+            buildDockerEnv('testing.Dockerfile').inside {
+                withEnv(environment) {
                     try {
-                        sh """
+                        sh '''
                            mkdir build-dir
                            cd build-dir
                            cmake -D CMAKE_BUILD_TYPE=RelWithDebInfo -D REALM_VALGRIND=ON -D REALM_ENABLE_ALLOC_SET_ZERO=ON -D REALM_MAX_BPNODE_SIZE=1000 -G Ninja ..
-                        """
+                        '''
                         runAndCollectWarnings(
                             script: 'cd build-dir && ninja',
                             name: "linux-valgrind",
                             filters: warningFilters,
                         )
-                        sh """
+                        sh '''
                             cd build-dir/test
                             valgrind --version
-                            valgrind --tool=memcheck --leak-check=full --undef-value-errors=yes --track-origins=yes --child-silent-after-fork=no --trace-children=yes --suppressions=${workspace}/test/valgrind.suppress --error-exitcode=1 ./realm-tests
-                        """
+                            valgrind --tool=memcheck --leak-check=full --undef-value-errors=yes --track-origins=yes --child-silent-after-fork=no --trace-children=yes --suppressions=$WORKSPACE/test/valgrind.suppress --error-exitcode=1 ./realm-tests
+                        '''
                     } finally {
-                        recordTests("Linux-ValgrindDebug")
+                        recordTests('Linux-RelWithDebInfo')
```

Comment:
we aren't running this right now, but wouldn't it be useful to mention something about valgrind in the label here?